sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -408,9 +417,17 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
         .map(visitIdentifierList(_).toArray)
         .getOrElse(Array.empty[String])
     val properties = Option(ctx.tableProps).map(visitPropertyKeyValues).getOrElse(Map.empty)
-    val bucketSpec = Option(ctx.bucketSpec()).map(visitBucketSpec)
+    val bucketSpec = if (ctx.bucketSpec().size > 1) {
+      duplicateClausesNotAllowed("CLUSTERED BY", ctx)
```

Comment:
Can you split the validation logic and the extraction logic? In this case I'd move the check to line 411 and do the extract on line 420.

Reply:
Sure